### PR TITLE
Initial commit to EWS Connector

### PR DESCRIPTION
**Pull request by tharindu1st**

### CODE REVIEW

File: src/main/java/org/wso2/carbon/connector/CreateAttachmentMediator.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        } catch (AxisFault axisFault) {
+            String msg = "Couldn't set SOAPEnvelope to MessageContext";
+            log.error(msg, axisFault);
+            throw new ConnectException(axisFault, msg);
+        } catch (TransformerException e) {
+            String msg = "Couldn't transform message";
+            log.error(msg, e);
+            throw new ConnectException(e, msg);
+        }
+
```
Comment (hariss63) [inline src/main/java/org/wso2/carbon/connector/CreateAttachmentMediator.java]: remove empty line. Do it in whole PR

File: src/main/java/org/wso2/carbon/connector/CreateAttachmentMediator.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    private SOAPBody populateBody(MessageContext messageContext) throws XMLStreamException, TransformerException {
+        SOAPBody soapBody = soapFactory.createSOAPBody();
+        OMElement createAttachment = soapFactory.createOMElement(EWSConstants.CREATE_ATTACHMENT_ELEMENT, message);
+        EWSUtils.populateItemIdAndChangeKeyAttributes(messageContext, createAttachment, EWSConstants
+                .PARENT_ITEM_ID_ELEMENT, EWSConstants.PARENT_ITEM_ID, message);
+        EWSUtils.populateDirectElements(messageContext, createAttachment, EWSConstants.ATTACHMENTS, message);
+        soapBody.addChild(createAttachment);
+        return soapBody;
+    }
+
```
Comment (hariss63) [inline src/main/java/org/wso2/carbon/connector/CreateAttachmentMediator.java]: remove empty line

File: src/main/java/org/wso2/carbon/connector/CreateItemMediator.java [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+package org.wso2.carbon.connector;
+
+
```
Comment (hariss63) [inline src/main/java/org/wso2/carbon/connector/CreateItemMediator.java]: remove empty line

File: src/main/java/org/wso2/carbon/connector/CreateItemMediator.java [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+                .IS_RESPONSE_REQUESTED_ELEMENT);
+        EWSUtils.setValueToXMLElement(messageContext, EWSConstants.REFERENCES, baseElement, EWSConstants
+                .REFERENCES_ELEMENT);
+        populateDirectElements(messageContext, baseElement, EWSConstants.REPLY_TO);
+        populateDirectElements(messageContext, baseElement, EWSConstants.RECEIVED_BY);
+        populateDirectElements(messageContext, baseElement, EWSConstants.RECEIVED_REPRESENTING);
+        populateDirectElements(messageContext, baseElement, EWSConstants.APPROVAL_REQUEST_DATA);
+        populateDirectElements(messageContext, baseElement, EWSConstants.VOTING_INFORMATION);
+        populateDirectElements(messageContext, baseElement, EWSConstants.REMINDER_MESSAGE_DATA);
+
```
Comment (hariss63) [inline src/main/java/org/wso2/carbon/connector/CreateItemMediator.java]: remove empty lines

File: src/main/java/org/wso2/carbon/connector/EWSConstants.java [hunk 5]
```diff
@@ -0,0 +1,242 @@
+package org.wso2.carbon.connector;
```
Comment (hariss63) [inline src/main/java/org/wso2/carbon/connector/EWSConstants.java:15]: Add the Header

File: src/main/java/org/wso2/carbon/connector/CreateItemMediator.java [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+        } catch (AxisFault axisFault) {
+            String msg = "Couldn't set SOAPEnvelope to MessageContext";
+            log.error(msg, axisFault);
+            throw new ConnectException(axisFault, msg);
+        } catch (TransformerException e) {
+            String msg = "Couldn't transform message";
+            log.error(msg, e);
+            throw new ConnectException(e, msg);
+        }
+
```
Comment (hariss63) [inline src/main/java/org/wso2/carbon/connector/CreateItemMediator.java]: remove empty line

File: src/main/java/org/wso2/carbon/connector/EWSConstants.java [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+    static final String SEEK_TO_CONDITION_PAGE_ITEM_VIEW = "seekToConditionPageItemView";
+    static final String CALENDAR_VIEW = "calendarView";
+    static final String CONTACTS_VIEW = "contactsView";
+    static final String GROUP_BY = "groupBy";
+    static final String DISTINGUISHED_GROUP_BY = "distinguishedGroupBy";
+    static final String RESTRICTION = "restriction";
+    static final String SORT_ORDER = "sortOrder";
+    static final String PARENT_FOLDER_IDS = "parentFolderIds";
+    static final String QUERY_STRING = "queryString";
+
```
Comment (hariss63) [inline src/main/java/org/wso2/carbon/connector/EWSConstants.java]: remove empty line

File: src/main/java/org/wso2/carbon/connector/EWSConstants.java [hunk 8]
```diff
@@ -0,0 +1,242 @@
...
+}
+
```
Comment (hariss63) [inline src/main/java/org/wso2/carbon/connector/EWSConstants.java]: remove empty lines

File: src/main/java/org/wso2/carbon/connector/FindItemMediator.java [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+        } catch (AxisFault axisFault) {
+            String msg = "Couldn't set SOAPEnvelope to MessageContext";
+            log.error(msg, axisFault);
+            throw new ConnectException(axisFault, msg);
+        } catch (TransformerException e) {
+            String msg = "Couldn't transform message";
+            log.error(msg, e);
+            throw new ConnectException(e, msg);
+        }
+
```
Comment (hariss63) [inline src/main/java/org/wso2/carbon/connector/FindItemMediator.java]: remove empty line

File: src/main/java/org/wso2/carbon/connector/FindItemMediator.java [hunk 10]
```diff
@@ -0,0 +1,145 @@
...
+            }
+            if (body != null){
+                queryStringElement.setText(body.getText());
+            }
+            findItemElement.addChild(queryStringElement);
+        }
+        soapBody.addChild(findItemElement);
+        return soapBody;
+    }
+
```
Comment (hariss63) [inline src/main/java/org/wso2/carbon/connector/FindItemMediator.java]: remove empty lines

File: src/test/java/org/wso2/carbon/connector/EWSIntegrationTest.java [hunk 11]
```diff
@@ -0,0 +1,127 @@
...
+ */
+public class EWSIntegrationTest extends ConnectorIntegrationTestBase {
+
+    private Map<String, String> esbRequestHeadersMap = new HashMap<String, String>();
+    private Map<String, String> namespaceMap = new HashMap<String, String>();
+    private String itemId, changeKey;
+
+    @BeforeClass(alwaysRun = true)
+    public void setEnvironment() throws Exception {
+        init("ews-connector-1.0.0");
```
Comment (RKathees) [inline src/test/java/org/wso2/carbon/connector/EWSIntegrationTest.java]: Can we the version from pom?

File: src/test/java/org/wso2/carbon/connector/EWSIntegrationTest.java [hunk 12]
```diff
@@ -0,0 +1,127 @@
...
+
+import javax.xml.namespace.QName;
+import java.io.File;
+import java.io.FileInputStream;
+import java.nio.charset.Charset;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Sample integration test
```
Comment (RKathees) [inline src/test/java/org/wso2/carbon/connector/EWSIntegrationTest.java]: Add meaningful description.

File: src/test/resources/artifacts/ESB/config/proxies/sendItemOperation.xml [hunk 13]
```diff
@@ -0,0 +1,80 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (RKathees) [inline src/test/resources/artifacts/ESB/config/proxies/sendItemOperation.xml:1]: Add wso2 license header in all proxies.

File: src/main/java/org/wso2/carbon/connector/EWSConstants.java (hunk 5 above)
Comment (tharindu1st) [reply src/main/java/org/wso2/carbon/connector/EWSConstants.java:15]: added

File: src/test/resources/artifacts/ESB/config/proxies/updateItemOperation.xml [hunk 14]
```diff
@@ -0,0 +1,78 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (hariss63) [inline src/test/resources/artifacts/ESB/config/proxies/updateItemOperation.xml:1]: The Header is missing

File: src/main/java/org/wso2/carbon/connector/DeleteItem.java [hunk 15]
```diff
@@ -0,0 +1,112 @@
...
+     */
+    private SOAPHeader populateSoapHeader(MessageContext messageContext) throws XMLStreamException,
+            TransformerException {
+        SOAPHeader soapHeader = soapFactory.createSOAPHeader();
+        EWSUtils.populateRequestedServerVersionHeader(soapHeader, messageContext);
+        EWSUtils.populateMailboxCulture(soapHeader, messageContext);
+        EWSUtils.populateExchangeImpersonationHeader(soapHeader, messageContext);
+        return soapHeader;
+    }
+
```
Comment (hariss63) [inline src/main/java/org/wso2/carbon/connector/DeleteItem.java:86]: remove the empty line

File: src/main/java/org/wso2/carbon/connector/GetItem.java [hunk 16]
```diff
@@ -0,0 +1,108 @@
...
+    private SOAPBody populateBody(MessageContext messageContext) throws XMLStreamException, TransformerException,
+            IOException {
+        SOAPBody soapBody = soapFactory.createSOAPBody();
+        OMElement getItemElement = soapFactory.createOMElement(EWSConstants.GET_ITEM_ELEMENT, message);
+        getItemElement.addChild(populateItemShape(messageContext));
+        getItemElement.addChild(populateItemIds(messageContext));
+        soapBody.addChild(getItemElement);
+        return soapBody;
+    }
+
```
Comment (hariss63) [inline src/main/java/org/wso2/carbon/connector/GetItem.java:107]: remove empty line

File: src/main/java/org/wso2/carbon/connector/UpdateItem.java [hunk 17]
```diff
@@ -0,0 +1,119 @@
...
+    private SOAPHeader populateSoapHeader(MessageContext messageContext) throws XMLStreamException,
+            TransformerException, IOException {
+        SOAPHeader soapHeader = soapFactory.createSOAPHeader();
+        EWSUtils.populateTimeZoneContextHeader(soapHeader, messageContext);
+        EWSUtils.populateRequestedServerVersionHeader(soapHeader, messageContext);
+        EWSUtils.populateMailboxCulture(soapHeader, messageContext);
+        EWSUtils.populateExchangeImpersonationHeader(soapHeader, messageContext);
+        return soapHeader;
+    }
+
```
Comment (hariss63) [inline src/main/java/org/wso2/carbon/connector/UpdateItem.java:86]: remove empty line

File: src/main/resources/createItem/createItem.xml [hunk 18]
```diff
@@ -0,0 +1,158 @@
...
+                                <initialDuration>2000</initialDuration>
+                                <progressionFactor>1.0</progressionFactor>
+                                <maximumDuration>3000</maximumDuration>
+                            </suspendOnFailure>
+                        </default>
+                    </endpoint>
+                </call>
+            </else>
+        </filter>
+
```
Comment (hariss63) [inline src/main/resources/createItem/createItem.xml]: remove empty line

File: src/test/resources/artifacts/ESB/config/proxies/createAttachmentOperation.xml [hunk 19]
```diff
@@ -0,0 +1,73 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (hariss63) [inline src/test/resources/artifacts/ESB/config/proxies/createAttachmentOperation.xml:1]: Add header

File: src/test/resources/artifacts/ESB/config/proxies/getAttachmentOperation.xml [hunk 20]
```diff
@@ -0,0 +1,85 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (hariss63) [inline src/test/resources/artifacts/ESB/config/proxies/getAttachmentOperation.xml:1]: add header

File: src/test/resources/artifacts/ESB/config/proxies/getItemOperation.xml [hunk 21]
```diff
@@ -0,0 +1,150 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (hariss63) [inline src/test/resources/artifacts/ESB/config/proxies/getItemOperation.xml:1]: add header

Comment (tharindu1st) [reply src/test/resources/artifacts/ESB/config/proxies/getItemOperation.xml:1]: done

File: src/test/resources/artifacts/ESB/config/proxies/sendItemOperation.xml (hunk 13 above)
Comment (tharindu1st) [reply src/test/resources/artifacts/ESB/config/proxies/sendItemOperation.xml:1]: done

File: src/main/java/org/wso2/carbon/connector/DeleteItem.java (hunk 15 above)
Comment (tharindu1st) [reply src/main/java/org/wso2/carbon/connector/DeleteItem.java:86]: done

File: src/main/java/org/wso2/carbon/connector/GetItem.java (hunk 16 above)
Comment (tharindu1st) [reply src/main/java/org/wso2/carbon/connector/GetItem.java:107]: done

File: src/main/java/org/wso2/carbon/connector/UpdateItem.java (hunk 17 above)
Comment (tharindu1st) [reply src/main/java/org/wso2/carbon/connector/UpdateItem.java:86]: done

File: src/test/resources/artifacts/ESB/config/proxies/updateItemOperation.xml (hunk 14 above)
Comment (tharindu1st) [reply src/test/resources/artifacts/ESB/config/proxies/updateItemOperation.xml:1]: done

File: src/test/resources/artifacts/ESB/config/proxies/getAttachmentOperation.xml (hunk 20 above)
Comment (tharindu1st) [reply src/test/resources/artifacts/ESB/config/proxies/getAttachmentOperation.xml:1]: done

File: src/test/resources/artifacts/ESB/config/proxies/createAttachmentOperation.xml (hunk 19 above)
Comment (tharindu1st) [reply src/test/resources/artifacts/ESB/config/proxies/createAttachmentOperation.xml:1]: done

**Pull request merged**